lib/utils.js

```diff
@@ -194,7 +194,16 @@ function prettifyLevel ({ log, colorizer = defaultColorizer }) {
  * key is not a string, then `undefined` will be returned. Otherwise, a string
  * that is the prettified message.
  */
-function prettifyMessage ({ log, messageKey = MESSAGE_KEY, colorizer = defaultColorizer }) {
+function prettifyMessage ({ log, messageFormat, messageKey = MESSAGE_KEY, colorizer = defaultColorizer }) {
+  if (messageFormat) {
+    const message = messageFormat.replace(/{([^}]+)}/g, function (match, p1) {
```

Comment:
Can you add some escaping support and tests?
So that somebody could use `\{{level}\} - \{{pid}\}`.

Reply:
Also, please update the docblock to describe the new parameter. Is it a required parameter? If not, what is the default?

Reply:
Can the `function (match, p1) {}` be defined as a top level utility so that it doesn't have to be recreated every time `prettifyMessage` is invoked?

Reply:
> Can you add some escaping support and tests?
> So that somebody could use `\{{level}\} - \{{pid}\}`.

Thanks for your review.
Since `\{' === '{'`, I will update the regex `/{([^}]+)}/` => `/{([^{}]+)}/` for supporting `{{level}}` => `{INFO}`.

I think there is no necessary for escaping, cause the format string must be `\\{{level}\\}`.

Reply:
> Can the `function (match, p1) {}` be defined as a top level utility so that it doesn't have to be recreated every time `prettifyMessage` is invoked?

The replacer function needs capture the log, may I ask how to hoist the function? Thanks.

Reply:
> Can the `function (match, p1) {}` be defined as a top level utility so that it doesn't have to be recreated every time `prettifyMessage` is invoked?

The replacer function needs capturing the log, may I ask how to hoist the function? Thanks.

Reply:
I'm fine with the function as-is. Moving it out of scope would require a `.bind`. And switching to an arrow function has no tangible benefit -- https://jsperf.com/anon-vs-arrow/1